third_party/inputmask/inputmask.date.extensions.js

```diff
@@ -78,7 +78,7 @@ export function factory(Inputmask) {
         return opts.tokenizer;
     }
     function isValidDate(dateParts, currentResult) {
-        return !isFinite(dateParts.rawday) || dateParts.day == "29" && !isFinite(dateParts.rawyear) || new Date(dateParts.date.getFullYear(), isFinite(dateParts.rawmonth) ? dateParts.month : dateParts.date.getMonth() + 1, 0).getDate() >= dateParts.day ? currentResult : false;
+        return !isFinite(dateParts.rawday) || dateParts.day == "29" && !isFinite(dateParts.rawyear) || dateParts.rawyear < 1000 || new Date(dateParts.date.getFullYear(), isFinite(dateParts.rawmonth) ? dateParts.month : dateParts.date.getMonth() + 1, 0).getDate() >= dateParts.day ? currentResult : false;
```

Comment:
I think, ideally, this would fail on the year if you did something like "02/29/0005". This would match the mask, since you have 4 year digits, but is not a valid leap year. I'm guessing from `dateParts.day`, would `dateParts.year` be `"0005"` in this case? If so, maybe you could instead check `dateParts.year.length < 4` instead?

Reply:
Good idea, done.